### PR TITLE
RUN --network=none support

### PR DESCRIPTION
**Pull request by alexcb**
Introduces ability to prevent programs from using network resources.

### CODE REVIEW

**Comment by alexcb** [comment]
implements #834

**Comment by alexcb** [comment]
#2718 needs to be merged first; then i'll ensure the `%s%s%s...` part is correct and remove my TODO comment.

File: earthfile2llb/converter.go [hunk 1]
```diff
@@ -1815,14 +1819,19 @@ func (c *Converter) internalRun(ctx context.Context, opts ConvertRunOpts) (pllb.
...
 	runOpts = append(runOpts, mountRunOpts...)
 	commandStr := fmt.Sprintf(
-		"%s %s%s%s%s%s%s%s",
+		"%s %s%s%s%s%s%s%s%s",
 		opts.CommandName, // e.g. "RUN", "IF", "FOR", "ARG"
 		strIf(opts.Privileged, "--privileged "),
 		strIf(opts.Push, "--push "),
 		strIf(opts.WithSSH, "--ssh "),
 		strIf(opts.NoCache, "--no-cache "),
+		strIf(opts.NoCache, "--network=none "),
```
Comment (vladaionescu) [inline earthfile2llb/converter.go:1834]:
```suggestion
		strIf(opts.NoNetwork, "--network=none "),
```

Comment (alexcb) [reply earthfile2llb/converter.go]: good catch! terrible copy/paste error :facepalm:

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**